tests/cupy_tests/math_tests/test_misc.py

```diff
@@ -255,6 +255,48 @@ def test_nan_to_num_broadcast(self, kwarg):
             with pytest.raises(ValueError):
                 xp.nan_to_num(0.0, **{kwarg: y})
 
+    @testing.for_all_dtypes(no_bool=True, no_complex=True)
+    @testing.numpy_cupy_array_equal()
+    def test_real_if_close(self, xp, dtype):
+        x = testing.shaped_random((10,), xp, dtype)
+        return xp.real_if_close(x)
+
+    @testing.numpy_cupy_array_equal()
+    def test_real_if_close_complex64(self, xp):
+        x = testing.shaped_random((10,), xp, dtype='F') + 1e-15j
```

Comment:
The imaginary part of `testing.shaped_random((10,), xp, dtype='F')` is not zeros.

Reply:
How about using `numpy.finfo.eps` to specify the error value?

Reply:
> The imaginary part of `testing.shaped_random((10,), xp, dtype='F')` is not zeros.

Ouch yes! My bad, will fix it.

Reply:
To avoid such failures, how about adding an assertion that compares out.dtype with x.dtype?